Apps/Sandcastle/gallery/development/Many Clipping Planes.html

```diff
@@ -0,0 +1,286 @@
+<div id="toolbar">
+    <table><tbody>
+        <tr>
+            <td>Distance</td>
+            <td>
+                <input type="range" min="-100.0" max="100.0" step="1.0" data-bind="value: cylinderRadius, valueUpdate: 'input'">
+                <input type="text" size="5" data-bind="value: cylinderRadius">
+            </td>
+            <td>Plane Count</td>
+            <td>
+                <input type="range" min="1" max="128" step="1" data-bind="value: planeCount, valueUpdate: 'input'">
+                <input type="text" size="5" data-bind="value: planeCount">
+            </td>
+        </tr>
+    </tbody></table>
+    <select data-bind="options: exampleTypes, value: currentExampleType"></select>
+</div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer', {
+    infoBox: false,
+    selectionIndicator: false,
+    shouldAnimate : true,
+    projectionPicker : true
+});
+
+viewer.terrainProvider = new Cesium.CesiumTerrainProvider({
+    url : 'https://assets.agi.com/stk-terrain/v1/tilesets/world/tiles',
+    requestWaterMask : true,
+    requestVertexNormals : true
+});
+viewer.extend(Cesium.viewerCesium3DTilesInspectorMixin);
+
+var globe = viewer.scene.globe;
+globe.depthTestAgainstTerrain = true;
+
+var cylinderRadius = -20.0;
+var radiusMultiplier = 1.0;
+
+var steps = 32;
+var clippingPlanes = [];
+var modelEntityClippingPlanes;
+var clippingModeUnion = false;
+var enabled = true;
+
+var clipObjects = ['model', 'b3dm', 'pnts', 'i3dm', 'terrain'];
+var viewModel = {
+    cylinderRadius : cylinderRadius,
+    exampleTypes : clipObjects,
+    currentExampleType : clipObjects[0],
+    planeCount : steps
+};
+
+Cesium.knockout.track(viewModel);
+
+var toolbar = document.getElementById('toolbar');
+Cesium.knockout.applyBindings(viewModel, toolbar);
+
+Cesium.knockout.getObservable(viewModel, 'cylinderRadius').subscribe(
+    function(newValue) {
+        cylinderRadius = parseFloat(viewModel.cylinderRadius);
+        updatePlanes();
+    }
+);
+
+Cesium.knockout.getObservable(viewModel, 'planeCount').subscribe(
+    function(newValue) {
+        var newSteps = parseFloat(viewModel.planeCount);
+        if (newSteps !== steps) {
+            steps = newSteps;
+            modelEntityClippingPlanes.removeAll();
+            computePlanes();
+        }
+    }
+);
+
+var scene = viewer.scene;
+var planeEntities = [];
+var selectedPlane;
+
+function updatePlanes() {
+    for (var i = 0; i < clippingPlanes.length; i++) {
+        var plane = clippingPlanes[i];
+        plane.distance = cylinderRadius * radiusMultiplier;
+    }
+}
+
+function computePlanes() {
+    var stepDegrees = 6.28319 / steps;
+    clippingPlanes = [];
+
+    for (var i = 0; i < steps; i++) {
+        var angle = i * stepDegrees;
+        var dir = new Cesium.Cartesian3();
+        dir.x = 1.0;
+        dir.y = Math.tan(angle);
+        if (angle > 1.57079632679) {
```

Comment:
Here and below, avoid hardcoding values.  There are `PI` constants here: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Math.js#L321